### PR TITLE
Refactor Task + Add mark features

### LINKED ISSUE #93: Add attributes to Task and sort TaskList
- Add a Status to Task: InProgress, Late, Complete
- Add a hidden attribute: `date` specifying its creation date solely for sorting purposes
- Sort Tasks according to its Status (Inprogress > Late > Complete), followed by its creation date

### LINKED ISSUE #96: Mark Tasks Feature
- Include new features: `markLate`, `markComplete` and `markInProgress` - format is similar to deleteTask

### PR DESCRIPTION
**Pull request by toh-xinyi**
# Changes
- Create new enum `TaskStatus` to hold all possible `Task` statuses: `INPROGRESS`, `LATE`, `COMPLETE`
- Add `status` and hidden `creationDate` attributes to `Task` class
- Make `Task implement Comparable<Task>` to compare `Task` objects based on its `status` followed by `creationDate`
- `TaskList` is sorted everytime a task is marked or a new task is added
- `markTaskComplete`, `markTaskLate` and `markTaskInProgress` commands implemented

### Other changes **TO BE ADDED BY WED TUTORIAL** 
- TEST CASES FOR new commands
- Update UG with new commands

---
# Issues
- Resolves #93 
- Resolves #96

### CODE REVIEW

**Comment by CarrieLi1015** [comment]
LGTM! Nice work :))

**Review by astraxq — CHANGES_REQUESTED** [review]
Hmm, I have yet to check the rest but I've realise quite a major bug:

- Adding tasks isn't reflected on the GUI, you might need to check on this.

File: src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public class DeleteTaskCommand extends Command {
      * a specified person
      */
     public DeleteTaskCommand(Index studentIndex, Index taskIndex) {
+        requireAllNonNull(studentIndex, taskIndex);
```
[the remark this answers is not in the document]
Comment (toh-xinyi) [reply src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java:38]: Good point. We can do so next week.

**Comment by toh-xinyi** [comment]
> Hmm, I have yet to check the rest but I've realise quite a major bug:
> 
> * Adding tasks isn't reflected on the GUI, you might need to check on this. [1st Image]
> * In addition, changing to another student doesn't reflect the change on the task list too. [2nd Image]
> 

@astraxq Oops, thanks for noticing, I forgot to change the name of an attribute in the fxml file. It should be fixed now. Thanks for your help

**Review by astraxq — APPROVED** [review]

**Review by QQH0828 — COMMENTED** [review]
LGTM! Good Work!

**Pull request merged**